source/common/http/codec_client.h

```diff
@@ -201,8 +210,17 @@ class CodecClient : Logger::Loggable<Logger::Id::client>,
     void onPreDecodeComplete() override { parent_.responseDecodeComplete(*this); }
     void onDecodeComplete() override {}
 
+    void onStreamMaxDurationReached() {
+      parent_.host_->cluster().stats().upstream_rq_max_duration_reached_.inc();
+      max_stream_duration_timer_->disableTimer();
```

Comment:
Also make sure this timer is disarmed in `deleteRequest` and make sure it's covered by a test.